### PR TITLE
Partial support for source-maps

### PR DESCRIPTION
**Pull request by scinos**
Partial support for sourcemaps. It won't change the name of the sourcemap, but it will change the content to point to the correct re-hashed source code.

### CODE REVIEW

**Comment by scinos** [comment]
@STRML what do you think?

File: test/test.js [hunk 1]
```diff
@@ -129,4 +129,23 @@ describe('OutputHash', () => {
                 expect(commons.source()).to.contain(hash);
             });
         }));
+
+    it.only('Works with sourcemaps', () => webpackCompile('sourcemap')
```
Comment (STRML) [inline test/test.js]: should be `it`, not `it.only()`

**Comment by STRML** [comment]
This is good, and a major pain point. It is of course impossible to make the hashes match but so long as they are unique and change when the file contents change, there is no reason they have to be match exactly. LGTM aside from the minor test issue.

**Comment by STRML** [comment]
🎉

**Pull request merged**